protocol/src/fschacha20poly1305.rs

```diff
@@ -44,14 +42,14 @@ impl std::error::Error for Error {
 /// FSChaCha20Poly1305 is used for message packets in BIP324.
 #[derive(Clone)]
 pub struct FSChaCha20Poly1305 {
-    key: [u8; 32],
+    key: Key,
     message_counter: u64,
 }
 
 impl FSChaCha20Poly1305 {
     pub fn new(key: [u8; 32]) -> Self {
         FSChaCha20Poly1305 {
-            key,
+            key: Key::new(key),
```

Comment:
Would we want to re-export the `Key` type and pass it by value into the constructor here or is this mostly for internal type sanity?